instrumentation/jetty-httpclient/jetty-httpclient-9.2/testing/src/main/groovy/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/AbstractJettyClient9Test.groovy

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_2
+
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.context.Context
+import io.opentelemetry.context.Scope
+import io.opentelemetry.instrumentation.test.base.HttpClientTest
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import org.eclipse.jetty.client.HttpClient
+import org.eclipse.jetty.client.api.ContentResponse
+import org.eclipse.jetty.client.api.Request
+import org.eclipse.jetty.client.api.Response
+import org.eclipse.jetty.client.api.Result
+import org.eclipse.jetty.http.HttpMethod
+import org.eclipse.jetty.util.ssl.SslContextFactory
+import org.junit.Rule
+import org.junit.rules.TestName
+import spock.lang.Shared
+
+import java.util.concurrent.TimeUnit
+
+abstract class AbstractJettyClient9Test extends HttpClientTest<Request> {
+
+  abstract HttpClient createStandardClient()
+
+  abstract HttpClient createHttpsClient(SslContextFactory sslContextFactory)
+
+
+  @Shared
+  def client = createStandardClient()
+  @Shared
+  def httpsClient = null
+
+  @Rule
+  TestName name = new TestName()
+
+  Request jettyRequest = null
+
+  def setupSpec() {
+
+    //Start the main Jetty HttpClient
+    client.start()
+
+    SslContextFactory tlsCtx = new SslContextFactory()
+//    tlsCtx.setExcludeProtocols("TLSv1.3")
+    httpsClient = createHttpsClient(tlsCtx)
+    httpsClient.setFollowRedirects(false)
+    httpsClient.start()
+  }
+
+  @Override
+  Request buildRequest(String method, URI uri, Map<String, String> headers) {
+
+    HttpClient theClient = uri.scheme == 'https' ? httpsClient : client
+
+    Request request = theClient.newRequest(uri)
+
+    HttpMethod methodObj = HttpMethod.valueOf(method)
+    request.method(methodObj)
+    request.timeout(CONNECT_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+
+    jettyRequest = request
+
+    return request
+  }
+
+  @Override
+  String userAgent() {
+    if (name.methodName.startsWith('connection error') && jettyRequest.getAgent() == null) {
```

Comment:
Does a test fail without this? It seems like a bug in our test if so

Reply:
Yes, Jetty 9.2 vs Jetty 9.4 react differently during the connection error testing with respect to the user-agent.